pkg/deploy/util/util.go

```diff
@@ -429,6 +434,48 @@ func DeploymentsForCleanup(configuration *deployapi.DeploymentConfig, deployment
 	return relevantDeployments
 }
 
+// WaitForRunningDeployerPod waits a given period of time until the deployer pod
+// for given replication controller is not running.
+func WaitForRunningDeployerPod(podClient kclient.PodsNamespacer, rc *api.ReplicationController, timeout time.Duration) (bool, error) {
+	podName := DeployerPodNameForDeployment(rc.Name)
+	if len(podName) == 0 {
+		return false, nil
+	}
+	pod, err := podClient.Pods(rc.Namespace).Get(podName)
+	if err != nil {
+		return false, err
+	}
+	if !controller.IsPodActive(*pod) {
+		return false, nil
+	}
+	watcher, err := podClient.Pods(rc.Namespace).Watch(
+		api.ListOptions{
+			FieldSelector:   fields.Set{"metadata.name": podName}.AsSelector(),
+			ResourceVersion: pod.ResourceVersion,
+		},
+	)
+	if err != nil {
+		return false, err
+	}
+	defer watcher.Stop()
+	if _, err := watch.Until(timeout, watcher, func(e watch.Event) (bool, error) {
+		if e.Type == watch.Error {
+			return false, fmt.Errorf("encountered error while watching for pod: %v", e.Object)
+		}
+		obj, isPod := e.Object.(*api.Pod)
+		if !isPod {
+			return false, errors.New("received unknown object while watching for pods")
+		}
+		if !controller.IsPodActive(*obj) {
```

Comment:
Shouldn't this be `if controller.IsPodActive(*obj) {` ? Otherwise, when it's running you will return false.